plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgrePrivilege.java

```diff
@@ -163,9 +163,16 @@ public short getPermission(PostgrePrivilegeType privilegeType) {
 
     public void setPermission(PostgrePrivilegeType privilegeType, boolean permit) {
         for (ObjectPermission permission : permissions) {
-            if (permission.privilegeType == privilegeType) {
+            if (permission.privilegeType != privilegeType) {
                 if (permit) {
                     permission.permissions |= GRANTED;
+                    ObjectPermission[] tempPermission = new ObjectPermission[this.permissions.length+1];
+                    for(int i = 0; i < this.permissions.length; i++) {
+                	tempPermission[i] = this.permissions[i];
+                    }
+                    tempPermission[this.permissions.length] = new ObjectPermission(privilegeType, permission.getGrantor(), permission.permissions);
+                    this.permissions = tempPermission;
+                    
```

Comment:
Can be replaced with `org.jkiss.utils.ArrayUtils#add`.

Reply:
`org.jkiss.utils.ArrayUtils#add` just add int arrays, i've a ObjectPermission array

Reply:
i change the for statement for `ArrayUtils.concatArrays()`

Reply:
There's generic overload; usage: `ArrayUtils.add(ObjectPermission.class, permissions, new ObjectPermission(...))`.

Please see my other commend below.